sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -333,14 +332,13 @@ protected AnalyzedRelation visitQuerySpecification(QuerySpecification node, Stat
 
         boolean isDistinct = node.getSelect().isDistinct();
         Symbol querySymbol = expressionAnalyzer.generateQuerySymbol(node.getWhere(), expressionAnalysisContext);
-        WhereClause whereClause = new WhereClause(querySymbol);
-        WhereClauseValidator.validate(whereClause.queryOrFallback());
+        WhereClauseValidator.validate(querySymbol == null ? Literal.BOOLEAN_TRUE : querySymbol);
         QueriedSelectRelation relation = new QueriedSelectRelation(
             isDistinct,
             List.copyOf(context.sources().values()),
             context.joinPairs(),
             selectAnalysis.outputSymbols(),
-            whereClause,
+            querySymbol == null ? Literal.BOOLEAN_TRUE : querySymbol,
```

Comment:
minor: we could store this into a local var to avoid checking for NULL twice

Reply:
@seut pushed a fixup to make `generateQuerySymbol` not return `null` in the first place.

Reply:
👍 even better!